src/Polly.Core/Retry/RetryResilienceStrategy.cs

```diff
@@ -53,6 +53,15 @@ protected internal override async ValueTask<Outcome<T>> ExecuteCore<TState>(Func
         {
             var startTimestamp = _timeProvider.GetTimestamp();
             var outcome = await StrategyHelper.ExecuteCallbackSafeAsync(callback, context, state).ConfigureAwait(context.ContinueOnCapturedContext);
+            try
+            {
+                context.CancellationToken.ThrowIfCancellationRequested();
```

Comment:
How about just checking for `context.CancellationToken.IsCancellationRequested()` and then constructing the exception manually?

Alternatively, you can do the same check inside `ExecuteCallbackSafeAsync` and it will apply everywhere where it is used.

Reply:
I tried to follow the same pattern what we already have for the `_timeProvider.DelayAsync`:

https://github.com/App-vNext/Polly/blob/4a44d0e79d75a203f6ec5ef3d0887c196d8fe110/src/Polly.Core/Retry/RetryResilienceStrategy.cs#L106-L113

But yes for sure, I can rewrite the code to avoid try catch. Do you want?

Reply:
But yes for sure, I can rewrite the code to avoid try catch. Do you want?

Let's try please. I distinctly remember that any try-catch block had some impact on the micro benchmarks.